tests/testsuite/update.rs

```diff
@@ -2037,11 +2030,19 @@ fn update_precise_breaking_shared_non_ws() {
     p.cargo("update -Zupdate-precise-breaking -p shared@0.1.0 --precise 0.2.0")
         .masquerade_as_nightly_cargo(&[])
         .with_status(101)
+        // The lockfile update is failing for the same reason a non-breaking
+        // update would. There is still a shared@0.1.0 here that didn't get
+        // upgraded (the transitive dependency within bar), and we are now
+        // asking the lockfile update to update it to 0.2.0, which it cannot do.
         .with_stderr_data(str![[r#"
```

Comment:
Like with `--breaking`, should we limit this to just the ones that can be upgraded?

Like other cases, this can be broken out into your task list

Reply:
I'm curious what you think the correct behaviour should be. We are upgrading `shared`, so we are not trying to keep it unchanged. So the update tries to do its thing, but fails. Anyway, I'll add this to the task list.

Reply:
This is 

> Can we limit updates more in case of breaking upgrades? See https://github.com/rust-lang/cargo/pull/14140#discussion_r1656927375 and the test update_precise_breaking_shared_non_ws.

See https://github.com/rust-lang/cargo/issues/12425#issuecomment-2186198258

Reply:
I've renamed the task in the hope that it has enough context to stand on its own

> Should cargo update --precise <breaking> only attempt to update direct dependencies (which can succeed) or also transitive dependencies (which will typically fail) See https://github.com/rust-lang/cargo/pull/14140#discussion_r1656927375 and the test update_precise_breaking_shared_non_ws.